lib/scrub/type_checker.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Scrub
+  class TypeChecker
+    attr_reader :organization, :new_types, :old_types, :message
+
+    def initialize(organization:, new_types: [], old_types: [])
+      @organization = organization
+      @new_types = new_types
+      @old_types = old_types.any? ? old_types : get_old_types
+    end
+
+    def validate
```

Comment:
Tests aside, if the caller is going to raise when this method returns `false` and then retrieve the exception message, I would consider just `raise @message` as the negative return value from this. Maybe do the logging here, too. That would potentially make less stuff in `ScrubRunner#check_new_types`.

Reply:
I've axed the `message` part and now it's just raising. Added a new error class.